### PR TITLE
[AppService] Onboard local context for app service

### PR DESCRIPTION
**Pull request by arrownj**
**Description**  

This PR is try to onboard local context feature for app service.

The work is still in progress. Tests for local context part will be added later.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from azure.cli.core.commands.parameters import (resource_group_name_type, get_location_type,
                                                 get_resource_name_completion_list, file_type,
                                                 get_three_state_flag, get_enum_type, tags_type)
+from azure.cli.core.local_context import LocalContextAttribute, SET, GET
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:13]: what's SET/GET? could you pls put them into "LocalContextActions"

**Comment by yonzhan** [comment]
local context

**Comment by panchagnula** [comment]
@btardif FYI
@yonzhan can you explain the purpose of this change - are we doing this for all Azure services or is AppService the first one to be onboarded here?

**Comment by yonzhan** [comment]
@yungezz to help answer your question.

**Comment by yungezz** [comment]
hi @panchagnula I requested your review on the change :) yes Azure CLI is doing local context for all services, webapp is first one. I'm starting an email thread to share more detail with plan involving both of you and our PM.

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 2]
```diff
@@ -544,13 +586,22 @@ def load_arguments(self, _):
...
 # Access Restriction Commands
     for scope in ['webapp', 'functionapp']:
-        with self.argument_context(scope + ' config access-restriction show') as c:
-            c.argument('name', arg_type=webapp_name_arg_type)
+        if scope == 'webapp':
+            with self.argument_context(scope + ' config access-restriction show') as c:
+                c.argument('name', arg_type=webapp_name_arg_type)
+        else:
+            with self.argument_context(scope + ' config access-restriction show') as c:
+                c.argument('name', arg_type=functionapp_name_arg_type)
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:596]:
```suggestion
        with self.argument_context(scope + ' config access-restriction show') as c:
            c.argument('name', arg_type=webapp_name_arg_type if scope =='webapp' else functionapp_name_arg_type)
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 1 above)
Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:13]: I defined it as CONSTANT currently, I will define a LocalContextAction enum later. As the change is in `core`, I want to leave this work to next sprint.

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py (hunk 2 above)
Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py]: thanks, updated.

**Pull request merged**